### PR TITLE
[WIP] Remove unnecessary explicit tests

### PR DESCRIPTION
**Pull request by danielmarbach**
Addresses https://github.com/Particular/NServiceBus/issues/3915

Many of the explicit tests were no longer necessary, outdated or don't make sense in the core repo. I think it is safe to remove those.

@SimonCropp My proposal is to turn ArgumentExceptionTests into a mandatory convention for core, thoughts?

// cc @adamralph

### CODE REVIEW

**Comment by adamralph** [comment]
Great!

BTW - there are still quite a few more of them which use the `Explicit` constructor which accepts a `string reason`. E.g. https://github.com/Particular/NServiceBus/blob/1ce80a0d0621b76d64b47177c664d3a45a4df0a3/src/NServiceBus.Core.Tests/AssemblyScanner/When_directory_with_no_reference_dlls_is_scanned.cs#L11

**Comment by SimonCropp** [comment]
> My proposal is to turn ArgumentExceptionTests into a mandatory convention for core, thoughts?

no complaints. that was always the long term intent, i just dropped the ball

**Comment by danielmarbach** [comment]
@SimonCropp any ideas how we would deal with things like

    public static class DateTimeExtensions
    {
       public static string ToWireFormattedString(DateTime dateTime)
        {
            return dateTime.ToUniversalTime().ToString(Format, CultureInfo.InvariantCulture);
        }
     }

We cannot properly guard against DateTime. Or would you recommend to guard against some date way in the past? Or should we just introduce a Guard.DoNothing method?

**Comment by danielmarbach** [comment]
@adamralph care to help out to introduce guards? If yes run the test ArgumentExceptionTest and fix the files highlighted. Maybe we could also improve the test a little bit by using an approval file?

**Comment by SimonCropp** [comment]
In min or max date valid?

**Comment by adamralph** [comment]
@danielmarbach will do.

File: src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs [hunk 1]
```diff
@@ -150,15 +153,14 @@ bool MethodCallSelf(MethodDefinition method)
             return false;
         }
 
-        static void WriteMethod(MethodDefinition method)
+        static void WriteMethod(MethodDefinition method, TextWriter builder)
```
Comment (adamralph) [inline src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs]: Would `writer` make more sense than `builder`?

Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs]: Change it as you wish

**Comment by adamralph** [comment]
Split into #3934 and #3936

**Pull request closed without merging**